### PR TITLE
initial check-in

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: src/main/java/org/eclipse/jdt/ls/debug/DebugEvent.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*******************************************************************************
```
Comment (akaroml) [inline src/main/java/org/eclipse/jdt/ls/debug/DebugEvent.java]: Please avoid checking in copies. Let's use symbolic link to reference source code from original repository.

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,69 @@
+apply plugin: 'java'
+apply plugin: 'eclipse'
+apply plugin: 'checkstyle'
+
+sourceSets {
+     main.java.srcDirs = ['src/main/java']
```
Comment (akaroml) [inline build.gradle]: Use symbolic link.

**Review by akaroml — CHANGES_REQUESTED** [review]

File: headless-src/test/java/org/eclipse/jdt/ls/debug/DebugUtilityTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        connectors.add(mockConnector);
+        Map<String, Argument> defaultArgumentsMap = new HashMap<String, Argument>();
+        defaultArgumentsMap.put("options", mockOptions);
+        defaultArgumentsMap.put("suspend", mockSuspend);
+        defaultArgumentsMap.put("main", mockMainClass);
+
+        EasyMock.expect(mockVMManager.launchingConnectors()).andReturn(connectors);
+        EasyMock.expect(mockConnector.defaultArguments()).andReturn(defaultArgumentsMap);
+
+        mockOptions.setValue("-cp c:/foo");
```
Comment (testforstephen) [inline headless-src/test/java/org/eclipse/jdt/ls/debug/DebugUtilityTest.java:76]: "-cp c:/foo" should not be hard code absolute path.

**Review by akaroml — CHANGES_REQUESTED** [review]
Please get rid of the binaries. These binaries need to be referenced from JDK, MVN Central or build from other projects. Please don't check in these files in source control.

**Comment by andxu** [comment]
@akaroml the source for jdi is not available in any mvn repo since JDK itself  is not built by MVN. We need the modified version of  language server bin and source on our cloned git, so it is not available in any MVN server.

**Comment by akaroml** [comment]
Can we reference %java_home%/lib/tools.jar instead of checking in the binary?

**Review by akaroml — CHANGES_REQUESTED** [review]
Can we reference %java_home%/lib/tools.jar instead of checking in the binary?

**Review by akaroml — APPROVED** [review]
Good job. Thanks.

**Pull request merged**